sources/Messages/Builders/FromArray.php

```diff
@@ -0,0 +1,54 @@
+<?php
+declare(strict_types=1);
+namespace Ciebit\ContactUs\Messages\Builders;
+
+use Ciebit\ContactUs\Messages\Message;
+use Ciebit\ContactUs\Status;
+use DateTime;
+
+class FromArray implements Builder
+{
+    private $data; #:array
+
+    public function setData(array $data): Builder
+    {
+        $this->data = $data;
+        return $this;
+    }
+
+    public function build(): Message
+    {
+        if (
+            ! is_array($this->data) OR
+            ! $this->data['name'] OR
+            ! $this->data['body'] OR
+            ! $this->data['email']
+        ) {
+            throw new Exception('ciebit.contactus.messages.builders.invalid', 3);
+        }
+        
+        $status = $this->data['status'] ? new Status((int) $this->data['status']) : Status::DRAFT();
```

Comment:
Aqui também poderá ocorrer um erro se o tipo de *status* for um objeto por exemplo.